src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -212,5 +214,35 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
                     break;
             }
         }
+
+        private JsonEncodedText FormatEnumValue(string value)
+        {
+            string converted;
+
+            if (!value.Contains(ValueSeparator))
```

Comment:
What if it contains commas that aren't followed by a space?

Reply:
This is processing the result of `Enum.ToString()`. I've looked at the implementation and it will return "0", one enum value, e.g. "Directory", or a comma-space delimited string, e.g. "Directory, Compressed".

Reply:
Ok, so we only support roundtripping data encoded with this exact converter?  i.e. if someone handcoded the json and didn't have a space after the comma, it's ok that it won't deserialize properly?  Just checking that's what we intend.

Reply:
@layomia Using `,` (without space) as the ValueSeparator and then trimming the elements would hit too hard the performance?

Reply:
> if someone handcoded the json and didn't have a space after the comma,

This code-path is for serialization where we format the result of Enum.ToString() on the enum value. User input is not processed here.

Reply:
I see.  Ok.  It's still a little fragile, but much less so.

How is any of this deserialized when using a naming policy other than default?

Reply:
> How is any of this deserialized when using a naming policy other than default?

We don't deserialize successfully in those scenarios. There's an open issue about if we should - https://github.com/dotnet/runtime/issues/31619. I think we should, if there's no significant perf hit. It's odd to serialize things we can't deserialize.

We'd have to cache the results of applying the naming policy to each enum value's name (similar to our caching for object properties), and compute the enum value by hand rather than [using `Enum.TryParse`](https://github.com/dotnet/runtime/blob/4eaccd61b1a2a4b078140f8b36f2b45ab5980e8e/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs#L59-L62).